### PR TITLE
Memberships: Remove unnecessary stripe disconnect success action

### PR DESCRIPTION
**Pull request by tyxla**
As @jsnajdr discovered in his review in https://github.com/Automattic/wp-calypso/pull/47460#discussion_r526888264 we can remove the `MEMBERSHIPS_CONNECTED_ACCOUNTS_STRIPE_DISCONNECT_SUCCESS` action in favor of the `MEMBERSHIPS_SETTINGS` action, which essentially does the same thing now that all the related state was removed in #47460. This PR takes care of that cleanup.

There should be no functional or visual changes introduced by this PR.

#### Changes proposed in this Pull Request

* Memberships: Remove unnecessary stripe disconnect success action

#### Testing instructions

* Verify that after disconnecting Stripe, the memberships settings are properly refetched.

### CODE REVIEW

File: client/state/memberships/settings/actions.js [hunk 1]
```diff
@@ -35,10 +31,7 @@ export const requestDisconnectStripeAccount = (
 		return wpcom.req
 			.get( `/me/connected_account/stripe/${ connectedAccountId }/disconnect` )
 			.then( () => {
-				dispatch( {
-					siteId,
-					type: MEMBERSHIPS_CONNECTED_ACCOUNTS_STRIPE_DISCONNECT_SUCCESS,
-				} );
+				dispatch( requestSettings( siteId ) );
 				dispatch( {
 					type: NOTICE_CREATE,
```
Comment (jsnajdr) [inline client/state/memberships/settings/actions.js:36]: btw this should use the `successNotice` action creator from _client/state/notices/actions_. There are many places in MailChimp/Memberships code that create the action objects directly.

**Review by jsnajdr — APPROVED** [review]
Thanks! :ship:

File: client/state/memberships/settings/actions.js (hunk 1 above)
Comment (tyxla) [reply client/state/memberships/settings/actions.js:36]: I agree, I thought I'd ignore it as it was unrelated, I will follow up on this in another PR. 👍

Comment (tyxla) [reply client/state/memberships/settings/actions.js:36]: Addressed in #47669

**Pull request merged**